aten/src/ATen/Utils.h

```diff
@@ -139,32 +139,30 @@ inline int64_t prod_intlist(ArrayRef<int64_t> list) {
 }
 
 /**
- * Utility function used in tensor implementations, which
- * supplies the default generator to tensors, if an input generator
- * is not supplied. The input Generator* is also static casted to
+ * Utility function to static cast input Generator* to
  * the backend generator type (CPU/CUDAGenerator etc.)
  */
 template <typename T>
-static inline T * get_generator_or_default(Generator * expr, Generator * defaultValue) {
-  if (!expr) {
-    expr = defaultValue;
-  }
+static inline T * check_generator(Generator expr) {
   if (T::device_type() == expr->device().type()) {
-    return static_cast<T*>(expr);
+    return static_cast<T*>(expr.get());
   }
   AT_ERROR("Expected a '", T::device_type(), "' device type for generator but found '", expr->device().type(), "'");
 }
 
 /**
- * Utility function to static cast input Generator* to
+ * Utility function used in tensor implementations, which
+ * supplies the default generator to tensors, if an input generator
+ * is not supplied. The input Generator* is also static casted to
  * the backend generator type (CPU/CUDAGenerator etc.)
  */
 template <typename T>
-static inline T * check_generator(Generator * expr) {
-  if (T::device_type() == expr->device().type()) {
-    return static_cast<T*>(expr);
+static inline T* get_generator_or_default(Generator expr, Generator defaultValue) {
```

Comment:
These can take the generator by reference, the function doesn't take ownership

Reply:
addressed